Source/MicrowaveNetworks/Touchstone/IO/TouchstoneReader.cs

```diff
@@ -277,6 +283,28 @@ private static List<string> TrimAndSplitLine(string line)
             string[] data = Regex.Split(line, @"\s+");
             return new List<string>(data);
         }
+        private bool TryParseImpedance(string impedance, out float r, out float x)
+        {
+            r = 0;
+            x = 0;
+            bool parsed = float.TryParse(impedance,out r);
+            if (!parsed)
+            {
+                if(impedance.Contains("j"))
+                {
+                    string value = Regex.Replace(impedance, @"[()j]", string.Empty);
+                    int sign = value.Contains("-") ? -1 : 1;
+                    string[] data = Regex.Split(value, @"[+-]");
+                    if (data.Length != 2) return false;
+                    bool parsedR = float.TryParse(data[0], out r);
+                    bool parsedX = float.TryParse(data[1], out x);
+                    x = x*sign;
+                    return parsedR && parsedX;
+                }
```

Comment:
This is just fine. As an alternative (if you like regular expressions 😆 ) , you could use [this regular expression I put together](http://regexr.com/6vam7). Lets you condense a few lines if you prefer to something like this (RegEx tested at the link above, code below not checked):

``` c#
// Expression testing: http://regexr.com/6vam7
Match m = Regex.Match(impedance, @"\((?<r>\d+)(?<sign>[+-])(?<x>\d+)j\)");
if (m.Success) 
{
   r = float.Parse(m.Groups["r"].Value);
   x = float.Parse(m.Groups["x"].Value);
   int sign = m.Groups["sign"].Value.Contains("-") ? -1 : 1;
   x *= sign;
}
return m.Success;
```

Reply:
Well, definitiely it is mor a riddle for non-regex experts but solvable if you want to to it. So  I am open to take yours.